carla/recourse_methods/catalog/face/library/face_method.py

```diff
@@ -0,0 +1,229 @@
+# import helpers
+import numpy as np
+
+# import Dijkstra's shortest path algorithm
+from scipy.sparse import csgraph, csr_matrix
+
+# import graph building methods
+from sklearn.neighbors import kneighbors_graph, radius_neighbors_graph
+
+
+def graph_search(
+    data,
+    index,
+    keys_mutable,
+    keys_immutable,
+    continuous_cols,
+    binary_cols,
+    model,
+    n_neighbors=50,
+    p_norm=2,
+    mode="knn",
+    frac=0.4,
+    radius=0.25,
+):
+    # This is our own implementation of these methods since the authors did not provide implementations online,
```

Comment:
I think we can leave this "explanation" inline comments out. If we really want to can put somelink like "(own implementation)" in the README instead?